src/main/java/io/strimzi/kafka/bridge/Application.java

```diff
@@ -119,15 +119,16 @@ private static Future<HttpBridge> deployHttpBridge(Vertx vertx, BridgeConfig bri
         Promise<HttpBridge> httpPromise = Promise.promise();
 
         HttpBridge httpBridge = new HttpBridge(bridgeConfig, metricsReporter);
-        vertx.deployVerticle(httpBridge, done -> {
-            if (done.succeeded()) {
-                LOGGER.info("HTTP verticle instance deployed [{}]", done.result());
-                httpPromise.complete(httpBridge);
-            } else {
-                LOGGER.error("Failed to deploy HTTP verticle instance", done.cause());
-                httpPromise.fail(done.cause());
-            }
-        });
+        vertx.deployVerticle(httpBridge)
+                .onComplete(done -> {
+                    if (done.succeeded()) {
+                        LOGGER.info("HTTP verticle instance deployed [{}]", done.result());
+                        httpPromise.complete(httpBridge);
+                    } else {
+                        LOGGER.error("Failed to deploy HTTP verticle instance", done.cause());
+                        httpPromise.fail(done.cause());
+                    }
+                });
```

Comment:
Here and elsewhere if we are using if (succeeeded) {} else {} why not use `onSuccess` and `onFailure` instead?
```suggestion
        vertx.deployVerticle(httpBridge)
                .onSuccess(result -> {
                        LOGGER.info("HTTP verticle instance deployed [{}]", result);
                        httpPromise.complete(httpBridge);
                }).onFailure(cause -> {
                        LOGGER.error("Failed to deploy HTTP verticle instance", cause);
                        httpPromise.fail(cause);
                });
```

Reply:
That's a good point. It's just kind of "old" code that we can refactoring the way you suggested.
Are you ok if I get this suggestion for a separate PR? It's not strictly related to the Vert.x 5 preparation.

Reply:
Yeah that makes sense